src/vcpkg/base/parse.cpp

```diff
@@ -90,14 +90,23 @@ namespace vcpkg
         }
     }
 
-    ParserBase::ParserBase(StringView text, StringView origin, TextRowCol init_rowcol)
+    ParserBase::ParserBase(StringView text, Optional<StringView> origin, TextRowCol init_rowcol)
         : m_it(text.begin(), text.end())
         , m_start_of_line(m_it)
         , m_row(init_rowcol.row_or(1))
         , m_column(init_rowcol.column_or(1))
         , m_text(text)
         , m_origin(origin)
     {
+#ifndef NDEBUG
+        if (auto check_origin = origin.get())
+        {
+            if (check_origin->empty())
+            {
+                Checks::unreachable(VCPKG_LINE_INFO, "origin should not be empty");
```

Comment:
Since this change I see this error:
~~~
$ /vcpkg format-manifest scripts/test_ports/vcpkg-ci-skia/vcpkg.json
/home/<me>/vcpkg-tool/src/vcpkg/base/parse.cpp(106): origin should not be empty
Aborted (core dumped)